src/evm/protocol/uniswap_v2/state.rs

```diff
@@ -98,6 +100,101 @@ impl ProtocolSim for UniswapV2State {
         cpmm_delta_transition(delta, reserve0_mut, reserve1_mut)
     }
 
+    fn swap_to_price(
+        &self,
+        token_in: &Bytes,
+        token_out: &Bytes,
+        target_price: Price,
+    ) -> Result<BigUint, SimulationError> {
+        let zero2one = token_in < token_out;
+        let (reserve_in, reserve_out) =
+            if zero2one { (self.reserve0, self.reserve1) } else { (self.reserve1, self.reserve0) };
+
+        // Flip target pool price to swap price
+        let swap_price_num = biguint_to_u256(&target_price.denominator);
+        let swap_price_den = biguint_to_u256(&target_price.numerator);
+
+        // Check reachability: target price must be above the spot price (with fees)
+        // swap_price_num/swap_price_den >= (reserve_in * FEE_PRECISION) / (reserve_out *
+        // FEE_NUMERATOR)
+        // Cross-multiply to avoid division: swap_price_num * reserve_out * FEE_NUMERATOR >=
+        // swap_price_den * reserve_in * FEE_PRECISION
+        let target_price_cross_mult = swap_price_num
+            .checked_mul(reserve_out)
+            .and_then(|x| x.checked_mul(FEE_NUMERATOR))
+            .ok_or_else(|| SimulationError::FatalError("Overflow in price check".to_string()))?;
+        let current_price_cross_mult = swap_price_den
+            .checked_mul(reserve_in)
+            .and_then(|x| x.checked_mul(FEE_PRECISION))
+            .ok_or_else(|| SimulationError::FatalError("Overflow in price check".to_string()))?;
+
+        if target_price_cross_mult < current_price_cross_mult {
+            return Ok(BigUint::ZERO);
+        }
+
+        // TODO: Detail arithmetic steps here.
+        // Calculate new reserve_in: x' = sqrt(k * price_num * FEE_NUMERATOR / (price_den *
+        // FEE_PRECISION))
+        let k = U512::from(reserve_in) * U512::from(reserve_out);
+        let k_times_price = k * U512::from(swap_price_num) * U512::from(FEE_NUMERATOR) /
+            (U512::from(swap_price_den) * U512::from(FEE_PRECISION));
+        let x_prime_u512 = sqrt_u512(k_times_price);
+
+        // Convert back to U256 and calculate amount_in
+        let limbs = x_prime_u512.as_limbs();
+        let x_prime = U256::from_limbs([limbs[0], limbs[1], limbs[2], limbs[3]]);
```

Comment:
@carloszanella Here should i just copy the docs from apex to detail the arithmetic steps to get to x' ?